mqtt/mqtt-edgehub/src/command.rs

```diff
@@ -58,81 +78,49 @@ impl CommandHandler {
             Duration::from_secs(60),
         );
 
-        CommandHandler {
+        let subscribe_topics = &[DISCONNECT_TOPIC.to_string()];
+        subscribe(&mut client, subscribe_topics).await?;
+
+        let (termination_handle, termination_receiver) = mpsc::channel(5);
+
+        Ok(CommandHandler {
             broker_handle,
             client,
-        }
+            termination_handle,
+            termination_receiver,
+        })
     }
 
     pub fn shutdown_handle(&self) -> Result<ShutdownHandle, ShutdownError> {
-        self.client
-            .shutdown_handle()
-            .map_or(Err(ShutdownError::ClientDoesNotExist), |shutdown_handle| {
-                Ok(ShutdownHandle(shutdown_handle))
-            })
+        Ok(ShutdownHandle(
+            self.termination_handle.clone(),
+            self.client.shutdown_handle()?,
+        ))
     }
 
-    pub async fn run(mut self) -> Result<(), CommandHandlerError> {
+    pub async fn run(mut self) -> Result<(), Error> {
```

Comment:
This function will poll the client, attempting to handle the events. If the client is disconnected, it will reconnect unless the command handler has been signaled to shutdown. This is an improvement because this structure allows for the client to reconnect if something weird happens inside the edgehub container.